### ISSUE TITLE
[URGENT]: Cookies contianing some special characters crash the WatsonWebsocketServer

### ISSUE THREAD

**Issue by WarstekHUN** [jchristn/WatsonWebsocket#119]
I've been experimenting the whole day and this is the result.
If the client has a cookie that includes some special characters, the following error occurs and the server stops:
```
[WatsonWsServer] listener exception:
System.ArgumentException: Specified value has invalid Control characters. (Parameter 'value')
   at System.Net.HttpValidationHelpers.CheckBadHeaderValueChars(String value)
   at System.Net.WebHeaderCollection.Add(String name, String value)
   at Interop.HttpApi.GetHeaders(IntPtr memoryBlob, IntPtr originalAddress)
   at System.Net.HttpListenerRequest.get_Headers()
   at System.Net.HttpListenerRequest.get_IsWebSocketRequest()
   at WatsonWebsocket.WatsonWsServer.AcceptConnections(CancellationToken cancelToken)
```
I'm sorry, I can't provide a list containing all these special characters

P.S.:
Also it would be so awesome if a ClientMetadata could have the clientside cookies. It would make for example loginToken checking much easier and efficient.

**Comment by jchristn** [comment]
Hi @WarstekHUN looking into this now

**Comment by jchristn** [comment]
Hi @WarstekHUN could you provide a sample cookie that reproduces this?  I just tried by adding a cookie on the client with the value ```tШЋЖ``` and was unable to reproduce it.

**Comment by jchristn** [comment]
Update: found this, https://stackoverflow.com/questions/22049796/specified-value-has-invalid-control-characters-when-converting-sha512-output-t

This is unfortunately something that is happening well beneath my library.

**Comment by jchristn** [comment]
Edit: however, you mention it's crashing - it shouldn't.  Looking at that now.

**Comment by jchristn** [comment]
Hi @WarstekHUN I just published v4.0.8 which has a fix in the ```AcceptConnections``` method to not exit when such a situation is encountered.  Please re-open if the issue persists or you encounter another problem!  Also added you to the README.

NuGet: https://www.nuget.org/packages/WatsonWebsocket/4.0.8
Commit: https://github.com/jchristn/WatsonWebsocket/commit/df03318326d31bd4c8fb2ef92455e247ef4cf947

**Comment by jchristn** [comment]
Thanks @WarstekHUN I wish I could implement changes in the underlay that this library uses.  Hopefully at least in this case the server doesn't crash on you :)  Cheers!  PS thank you so much for letting me know!
